library/core/src/cmp.rs

```diff
@@ -3,14 +3,17 @@
 //! This module contains various tools for comparing and ordering values. In
 //! summary:
 //!
-//! * [`Eq`] and [`PartialEq`] are traits that allow you to define total and
-//!   partial equality between values, respectively. Implementing them overloads
-//!   the `==` and `!=` operators.
+//! * [`Eq`] and [`PartialEq`] correspond to equivalence relations and partial
+//!   equivalence relations, respectively. Implementing these traits for types
```

Comment:
I find this wording confusing/misleading, since `T: PartialEq<U>` doesn't correspond to the usual notion of a partial equivalence relation. Only `T: PartialEq<T>` does.

Reply:
Would you have an idea how to phrase that better?

Maybe the following:

```
* [`PartialEq`] overloads the `==` and `!=` operators.
* [`Eq`] indicates that the overloaded `==` operator corresponds to an equivalence relation.
```

Reply:
These are on the module overview (`std::cmp`). Here, `T` isn't defined/explained. I could write `PartialEq<T>`, but this would be inconsistent with the remaining style in that section. So I'm a bit unsure what to do.

Reply:
FWIW the next bullet with PartialOrd has the same problem.

But the current docs are inconsistent, since the module-level docs say `PartialEq` is (always) a PER (partial equivalence relation) but the trait docs say something else.

Reply:
> These are on the module overview (std::cmp). Here, T isn't defined/explained. I could write `PartialEq<T>`, but this would be inconsistent with the remaining style in that section. So I'm a bit unsure what to do.

Happy to put my thumb on this scale: Please favor accuracy over consistency with styles. Right now I'd rather this was correct and looked like it got chewed up and spat out by five different math-gremlins, than that it was beautiful but misleading.

Reply:
> If you want to say what it corresponds to, you can also say
> 
> ```
> * [`PartialEq`] overloads the `==` and `!=` operators. If `T: PartialEq` (i.e., `Rhs` is `Self`), this corresponds to a partial equivalence relation.
> * [`Eq`] indicates that the overloaded `==` operator corresponds to an equivalence relation.
> ```

I feel like it needs an explanation what's `T`, so in fixup 981cff3ae8d66189ddfbc736a2cae2ace76b0d93, I propose:

```
//! * [`PartialEq<Rhs>`] overloads the `==` and `!=` operators. In cases where
//!   `Rhs` (the right hand side's type) is `Self`, this corresponds to a
//!   partial equivalence relation.
//! * [`Eq`] indicates that the overloaded `==` operator corresponds to an
//!   equivalence relation.
```

But feel free to rephrase if you think this is too verbose.

Reply:
This seems good. We can refine this for concision later.

Reply:
In fixup b71d6bc32ad81eca6b116574ed1634675e67b258, I replaced "this" with "this trait" to be more clear.